designcompose/src/main/java/com/android/designcompose/DocServer.kt

```diff
@@ -314,6 +340,11 @@ internal fun DocServer.fetchDocuments(
 
                 // Remember the new document
                 synchronized(documents) { documents[id] = doc }
+                synchronized(DesignSettings.designDocStatuses) {
```

Comment:
I don't think we're consistently locking `designDocStatuses` when updating it; I don't think reads from `designDocStatuses` by client code are synchronized at all (unless Kotlin adds that invisibly).

Reply:
Kotlin adds that, but yeah, I'm not sure, it seemed like a better safe than sorry thing since we're mixing threads and coroutines.

Reply:
I think Ralph is saying that the other places where you set variables in `DesignSettings.designDocStatuses[id]` are not synchronized.

Reply:
I see what he means now, I read it too quickly. I'm still not happy with how it's set up, we should talk offline about it.